### PR TITLE
feat: Game Version column in catalog

### LINKED ISSUE #216: Catalog columns
**Is your feature request related to a problem? Please describe.**
some addons on curse are not no longer maintained, so downloading them will slow/lag and some time crash the game, not to mention they wont work.

**Describe the solution you'd like**
Adding more columns in the catalog window, the same way it is in my addons window. this way if  an addon is no longer maintained i can see that from the addon game version.

### PR DESCRIPTION
**Pull request by casperstorm**
Resolves #216 #247 #230 

## Proposed Changes
  - Add Game Version as a column to Catalog.
  - Optional columns in Catalog.

## Checklist

- [x] Tested on all platforms changed
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

**Comment by casperstorm** [comment]
@tarkah if you have 5min at somepoint, can you run this branch and see what the hell is up with the resizing of the rows in catalog? I think I must doing something wrong. No rush on it, caching is more important.

File: src/gui/mod.rs [hunk 1]
```diff
@@ -764,7 +772,7 @@ impl Default for HeaderState {
                 ColumnState {
                     key: ColumnKey::LocalVersion,
                     btn_state: Default::default(),
-                    width: Length::Units(150),
+                    width: Length::Fill,
```
Comment (tarkah) [inline src/gui/mod.rs:775]: Why are we setting this as `Fill`? Shouldn't we only have one as Fill, everything else is either the default value or whatever value is the result after a resize operation?

File: src/gui/mod.rs [hunk 2]
```diff
@@ -832,7 +840,7 @@ pub struct ColumnState {
 
 impl From<&ColumnState> for ColumnConfigV2 {
     fn from(column: &ColumnState) -> Self {
-        // Only `ColumnKey::Title` should be saved as Length::Fill -> width: None
+        // Only `ColumnKey::Description` should be saved as Length::Fill -> width: None
```
Comment (tarkah) [inline src/gui/mod.rs:843]: It's still `ColumnKey::Title`?

File: src/gui/update.rs [hunk 3]
```diff
@@ -145,8 +193,29 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
                         }
                     });
```
Comment (tarkah) [inline src/gui/update.rs:195]: We're missing the `ajour.catalog_header_state.columns.iter()` here, which is probably messing things up

**Comment by tarkah** [comment]
@casperstorm caught the couple tiny issues that were holding it up. Took me a while to catch because everything starts looking the same 😵 Great stuff!

File: src/gui/update.rs [hunk 4]
```diff
@@ -85,25 +85,6 @@ pub fn handle_message(ajour: &mut Ajour, message: Message) -> Result<Command<Mes
...
-                                    None
-                                }
-                            })
-                            .next()
-                        {
-                            a.width = column.width.map_or(Length::Fill, Length::Units);
-                            a.hidden = column.hidden;
-                            a.order = idx;
-                        }
-                    });
```
Comment (tarkah) [inline src/gui/update.rs]: `ColumnConfig::V2` is from before we implemented the catalog, so this config only relates to the MyAddons stuff

**Comment by casperstorm** [comment]
> @casperstorm caught the couple tiny issues that were holding it up. Took me a while to catch because everything starts looking the same 😵 Great stuff!

Thanks for the assistance @tarkah and yeah its hard to get lost with this naming. Maybe we could consider moving it to a seperate file as well, eventually.

**Comment by casperstorm** [comment]
@tarkah could you give it one more look on the review side? I think im done now :)

**Pull request merged**